NGitLab.Mock/LabelsCollection.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Linq;
+
+namespace NGitLab.Mock
+{
+    public sealed class LabelsCollection : Collection<Label>
+    {
+        public LabelsCollection(GitLabObject parent)
+            : base(parent)
+        {
+        }
+
+        public Label GetById(int id)
+        {
+            return this.FirstOrDefault(mr => mr.Id == id);
+        }
+
+        public Label Add(string name = null, string color = null, string description = null)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                name = Guid.NewGuid().ToString();
+            }
+
+            if (string.IsNullOrEmpty(color))
+            {
+                color = "#d9534f";
+            }
+
+            var label = new Label
+            {
+                Name = name,
+                Color = color,
+                Description = description,
+            };
+
+            Add(label);
+            return label;
+        }
+
+        public override void Add(Label label)
+        {
+            if (label is null)
+                throw new ArgumentNullException(nameof(label));
+
+            if (label.Id == default)
+            {
+                label.Id = Server.GetNewLabelId();
+            }
+            else if (GetById(label.Id) != null)
```

Comment:
Are labels unique by name?

Reply:
I don't know, it's not specified in the [doc](https://docs.gitlab.com/ee/api/labels.html#create-a-new-label).

Reply:
I've tested using the web interface and you get an error : `Title has already been taken`